### PR TITLE
negative Shift removed, forloop removed to speedup hyperoptio

### PR DESCRIPTION
**Pull request by mablue**
live checked
Screenshot from 2021-08-21 02-19-36

### CODE REVIEW

**Comment by Kroissan** [comment]
image
it's buying now :) Will let this dry run over night, i'm also hyperopting it. I will let u know how it goes tomorrow.

File: user_data/strategies/Diamond.py [hunk 1]
```diff
@@ -32,82 +26,65 @@
...
+    trailing_stop_positive = 0.222
+    trailing_stop_positive_offset = 0.284
     trailing_only_offset_is_reached = True
+
     # Buy hypers
-    timeframe = '5m'
+    timeframe = '4h'
     # #################### END OF RESULT PLACE ####################
-    buy_push = DecimalParameter(0, 2, decimals=3, default=1, space='buy')
-    buy_shift = IntParameter(-10, 0, default=-6, space='buy')
```
Comment (xmatthias) [inline user_data/strategies/Diamond.py:83]: in theory, you could keep the shift (if it makes sense to do so) - but make the shift positive only.

**Comment by Kroissan** [comment]
> image

dumb me hyperopt it with 5m timeframe. it gave good result tho. Should it still be hyperopted with ShortTradeDurHyperOptLoss ?
Wouldnt it work better with  SharpeHyperOptLoss class ?

**Comment by mablue** [comment]
> > image
> 
> dumb me hyperopt it with 5m timeframe. it gave good result tho. Should it still be hyperopted with ShortTradeDurHyperOptLoss ?
> Wouldnt it work better with SharpeHyperOptLoss class ?

I think it will be beautiful that the name of strategy have a connection with his code!
I mean I changed Diamond fully! and now its really pure. no indicator calculate inside it. It just use the olhcv data for find a way to trade...
If you have a comment after compair this two I will be happy!

File: user_data/strategies/Diamond.py [hunk 2]
```diff
@@ -9,12 +9,36 @@
...
+# *    2/10:     10 trades. 7/0/3 Wins/Draws/Losses. Avg profit   5.50%. Median profit   7.05%. Total profit  0.01817970 BTC (  18.18%). Avg duration 0:27:00 min. Objective: -11.72450
+# freqtrade hyperopt --hyperopt-loss SortinoHyperOptLossDaily --spaces buy sell roi trailing stoploss --strategy Diamond -j 2 -e 10
+#   | * Best |    3/10 |      165 |     98   63    4 |        1.00% |    0.05453885 BTC   (54.54%) | 0 days 08:02:00 |    0.00442974 BTC   (13.41%) |     -41.371 |
+#   | * Best |    7/10 |      101 |     56   42    3 |        0.73% |    0.02444518 BTC   (24.45%) | 0 days 13:08:00 |    0.00107122 BTC    (3.24%) |    -66.7687 |
+# *    7/10:    101 trades. 56/42/3 Wins/Draws/Losses. Avg profit   0.73%. Median profit   0.13%. Total profit  0.02444518 BTC (  24.45%). Avg duration 13:08:00 min. Objective: -66.76866
+# freqtrade hyperopt --hyperopt-loss OnlyProfitHyperOptLoss --spaces buy sell roi trailing stoploss --strategy Diamond -j 2 -e 10
+# *    7/10:    117 trades. 74/41/2 Wins/Draws/Losses. Avg profit   1.91%. Median profit   1.50%. Total profit  0.07370921 BTC (  73.71%). Avg duration 9:26:00 min. Objective: -0.07371
```
Comment (Kroissan) [inline user_data/strategies/Diamond.py:40]: i really wonder how you get there Oo 73% is incredible. I get the best result with OnlyProfitHyperOptLoss, also with low max open trade like 3xunlimited (using stable coin for staking). 
I think i will do a 10k epoch tonight (lets use that 10900k). Then i'll dry run it for some days and tell you how it goes.

Comment (mablue) [reply user_data/strategies/Diamond.py:40]: try to run all commands by sorted order on this file its reached to 73 in 7t epoch!!!!!

**Review by xmatthias — APPROVED** [review]

**Pull request merged**